core/src/main/java/org/apache/iceberg/IcebergMetadata.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.util.PropertyUtil;
+import org.immutables.value.Value;
+import javax.annotation.Nullable;
+
+public interface IcebergMetadata extends Serializable {
```

Comment:
Can we name it something like BaseMetadata, and add javadoc?  IcebergMetadata seems it can refer to anything.

Also had a question, we don't need to move all of these for #9461 right?  Do you think it's easier if we just move the necessary ones like location, and properties()?

Reply:
I think only these four are mandatory to have common code for table and view. 
```
@Value.Parameter(order = 1003)
  String location();

  @Value.Parameter(order = 1008)
  Map<String, String> properties();

  @Value.Parameter(order = 1010)
  @Nullable
  String metadataFileLocation();

  @Value.Parameter(order = 1011)
  Schema schema();
```

I have already open this [PR](https://github.com/apache/iceberg/pull/9682) with iceberg .

Should I keep only the above one?